pkg/controller/mpi_job_controller.go

```diff
@@ -1004,9 +1009,20 @@ func (c *MPIJobController) updateMPIJobStatus(mpiJob *kubeflow.MPIJob, launcher
 				mpiJob.Status.CompletionTime = launcher.Status.CompletionTime
 			}
 			updateMPIJobConditions(mpiJob, kubeflow.JobSucceeded, corev1.ConditionTrue, mpiJobSucceededReason, msg)
+			
 			mpiJobsSuccessCount.Inc()
+			
+			if mpiJob.Status.StartTime != nil && mpiJob.Status.CompletionTime != nil {
+				duration := mpiJob.Status.CompletionTime.Sub(mpiJob.Status.StartTime.Time).Seconds()
+				mpiJobsDurationHistogram.WithLabelValues(launcher.Name, mpiJob.Namespace, "status: job success").Observe(duration)
+			}
 		} else if isJobFailed(launcher) {
 			c.updateMPIJobFailedStatus(mpiJob, launcher, launcherPods)
+
+			if mpiJob.Status.StartTime != nil && mpiJob.Status.CompletionTime != nil {
+				duration := mpiJob.Status.CompletionTime.Sub(mpiJob.Status.StartTime.Time).Seconds()
+				mpiJobsDurationHistogram.WithLabelValues(launcher.Name, mpiJob.Namespace, "status: job failure").Observe(duration)
+			}
```

Comment:
L1015-L1018 seems similar to L1022-L1025. Maybe, only the status message is different?
Can you commonize the process?